### DISCUSSION TITLE
백앤드 배포후 could not prepare statement  에러

### DISCUSSION THREAD

**Discussion by sagada** [fsoftwareengineer/todo-application#95] [질문과 답변] [answered]
1)에러 페이지 : 369p
2)해본것들 : UserEntity 확인하였는데 본코드랑 같음, gradle 의존성에 h2 뺏다가 넣다가 해봄
yml 파일 prod, dev 분리 확인 
스프링 버전 2.5.0
http://backtodo.ap-northeast-2.elasticbeanstalk.com/auth/signup)

헤당 api 호출 배포후에 에러가 납니다. 어떤 원인이 있을 수 있을까요?

에러 메시지 
{
    "error": "could not prepare statement; SQL [select userentity0_.id as col_0_0_ from user_entity userentity0_ where userentity0_.email=? limit ?]; nested exception is org.hibernate.exception.SQLGrammarException: could not prepare statement",
    "data": null
}

**Comment by fsoftwareengineer** [comment, marked as answer]
UserRepository 코드를 남겨 주시면 확인 해 보도록 하겠습니다.

**Reply by sagada** [reply]
@Data
@Entity
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(uniqueConstraints = {@UniqueConstraint(columnNames = "email")}, name = "user_entity")
public class UserEntity {
    @Id
    @GeneratedValue(generator = "system-uuid")
    @GenericGenerator(name = "system-uuid", strategy = "uuid")
    private String id;

    @Column(nullable = false)
    private String username;

    @Column(nullable = false)
    private String email;

    @Column(nullable = false)
    private String password;

}

**Reply by fsoftwareengineer** [reply]
TodoRepository와 비교했을 때 클래스 상단에 `@Repository` 가 안보이는데, 추가 하고 실행 해도 같은 결과가 나오는지 확인 해 주시기 바랍니다.

**Reply by sagada** [reply]
@Repository 를 추가해도 같은 결과입니다.

**Reply by sagada** [reply]
<img width="1078" alt="스크린샷 2022-03-12 오후 10 14 06" src="https://user-images.githubusercontent.com/59609682/158019347-2e712190-621c-41fe-ad4b-91b0c775578b.png">

해결 했습니다.
log 보니간 h2 데이터베이스가 적용된것을 확인했습니다. application-prod.yml 파일을 prod환경으로 인식 못한것 같아서
<img width="568" alt="스크린샷 2022-03-12 오후 10 14 57" src="https://user-images.githubusercontent.com/59609682/158019375-bba3a883-feb1-4cc0-a781-c7cd36d78a0d.png">

yml 파일을 위처럼 수정하였더니 정상 동작 합니다.
확인해주셔서 감사합니다 ^^
